src/main/client_context.cpp

```diff
@@ -73,8 +74,39 @@ struct ActiveQueryContext {
 	BaseQueryResult *open_result = nullptr;
 };
 
+#ifdef DEBUG
+struct DebugClientContextState : public ClientContextState {
+	~DebugClientContextState() override {
+		D_ASSERT(active_transaction == 0);
+		D_ASSERT(active_query == 0);
+	}
+
+	idx_t active_transaction = 0;
+	idx_t active_query = 0;
+
+	void QueryBegin(ClientContext &context) override {
+		active_query++;
```

Comment:
Can we assert here (and in `QueryEnd`) that `active_transaction == 1`? I.e. queries only exist in the context of an active transaction.